torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -347,6 +354,198 @@ at::Tensor tensor_from_cuda_array_interface(PyObject* obj) {
       at::device(kCUDA).dtype(dtype)
   );
 }
+
+
+template <typename T>
+std::string get_vector_str(std::vector<T> & vec, int64_t start, int64_t end=-1){
+  std::string output = "(";
+  if (end < 0) {
+    end = vec.size();
+  }
+  for (int i=start; i<end; ++i) {
+    output += (std::to_string(vec[i]) + ", ");
+  }
+  return output + ")";
+}  
+
+template <typename T>
+std::string get_buffer_str(void* buffer, int64_t start, int64_t end){
+  T* array = static_cast<T*>(buffer); 
+  std::string output = "(";
+  for (int i=start; i<end; ++i) {
+    output += (std::to_string(array[i]) + ", ");
+  }
+  return output + ")";
+}  
+
+
+bool _extract_ndarrays(PyObject* obj, std::vector<int64_t> & shape, size_t (&numels)[], size_t shape_idx, std::vector<PyArrayObject*> & array_ptr_storage) {
+  if (PyArray_Check(obj)) {
+    if (numels[shape_idx] != PyArray_SIZE((PyArrayObject*) obj)) {
+      auto true_shape_str = get_buffer_str<npy_intp> (static_cast<void*>(PyArray_DIMS((PyArrayObject*) obj)), 0, PyArray_NDIM((PyArrayObject*) obj)) + std::string(")");
+      auto expected_shape_str = get_vector_str(shape, shape_idx);
+      auto err_str = std::string("expected numpy array of shape ") + expected_shape_str + std::string(" at dim ") + std::to_string(shape_idx) + std::string("(got ") + true_shape_str;
+      throw ValueError(err_str);  
+    }
+    array_ptr_storage.push_back((PyArrayObject*) obj);
+    return true;
+  } 
+  else if (PySequence_Check(obj)) {
+    auto seq = THPObjectPtr(PySequence_Fast(obj, "not a sequence"));
+    if (!seq) throw python_error();
```

Comment:
Egyptian-style curly braces for every if/for statement so the linter doesn't complain